develop/tutorials/articles/290-configuration/02-custom-system-settings-ui.markdown

```diff
@@ -0,0 +1,226 @@
+                Map<String, Object> params = new HashMap<>();
+
+                String[] mysymbols = ParamUtil.getParameterValues(request, "mysymbols");
+
+                params.put("symbols", mysymbols);
+
+                return params;
+            }
+        }
+
+The above example generates a custom rendering (HTML) for the form in the
+`render()` method and reads the information entered in the custom form in the
+`getRequestParameters()` method.
+
+## Creating a Completely Custom Configuration UI
+
+In some cases, you want a completely custom UI for your configuration. For
+example: 
+
+- Your application doesn't use Config Admin to provide its configuration. You
+    have a completely different configuration backend, and you'll write a
+    completely independent frontend.
+
+- Your application needs more flexibility in its UI, such as multiple
+    configuration screens.
+
+To accomplish this, write a `ConfigurationScreen` implementation.
+
+At a high level you must
+
+1.  Write a Component that declares itself an implementation of the
+    `ConfigurationScreen` interface.
+
+2.  Implement `Configurationscreen`'s methods.
+
+3.  Create the UI by hand.
+
+Here's an example implementation:
+
+    @Component(immediate = true, service = ConfigurationScreen.class) 
+    public class SampleConfigurationScreen implements ConfigurationScreen {
+
+First declare the class an implementation of `ConfigurationScreen`.
+
+    @Override 
+    public String getCategoryKey() { 
+
+        return "third-party"; 
+        
+    }
+
+    @Override 
+    public String getKey() { 
+    
+        return "sample-configuration-screen"; 
+        
+    }
+
+    @Override 
+    public String getName(Locale locale) { 
+    
+        return "Sample Configuration Screen"; 
+
+    }
+
+Second, set the category key, the configuration entry's key, and its localized
+name. This example puts the configuration entry, keyed
+`sample-cofniguration-screen`, into the `third-party` System Settings section.
+The String that appears in System Settings is _Sample Configuration Screen_.
+
+    @Override 
+    public String getScope() { 
+    
+        return "system"; 
+        
+    }
+
+Third, set the 
+[configuration scope](/develop/tutorials/-/knowledge_base/7-1/scoping-configurations).
+
+    @Override 
+    public void render(HttpServletRequest request, HttpServletResponse response) 
+        throws IOException {
+
+        _jspRenderer.renderJSP( _servletContext, request, response,
+        "/sample_configuration_screen.jsp"); 
+        
+    }
+
+    @Reference private JSPRenderer _jspRenderer;
+
+    @Reference(
+        target ="(osgi.web.symbolicname=com.liferay.currency.converter.web)", 
+        unbind = "-")
+    private ServletContext _servletContext;
+
+The most important step is to write the `render` method.This example relies on
+the `JSPRenderer` service to delegate rendering to a JSP.
+
+It's beyond the scope of this tutorial to write the JSP markup. A separate
```

Comment:
@drewbrokke @nealrobbins I propose saying this, or some such thing. We should, of course, immediately follow up with the next tutorial. What do you guys think?

Reply:
That could work.  Essentially it's the same as writing jsps for an MVCRenderCommand, so there would be a lot of overlap.  However, the main caveat I'd note somewhere (not necessarily here) is that if they write an MVCActionCommand to go with their JSP, they must register it with the System Settings portlet.  I did this in my sample JSP, if you need a reference.